sql/src/main/java/io/crate/operation/aggregation/impl/PercentileAggregation.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.operation.aggregation.impl;
+
+import com.google.common.base.Preconditions;
+import com.tdunning.math.stats.AVLTreeDigest;
+import com.tdunning.math.stats.Centroid;
+import io.crate.Streamer;
+import io.crate.analyze.symbol.Function;
+import io.crate.breaker.RamAccountingContext;
+import io.crate.exceptions.CircuitBreakingException;
+import io.crate.metadata.DynamicFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.operation.Input;
+import io.crate.operation.aggregation.AggregationFunction;
+import io.crate.types.*;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+
+public class PercentileAggregation extends AggregationFunction<PercentileAggregation.TDigestState, Object> {
+
+    public static final String NAME = "percentile";
+    private static final int COMPRESSION = 100;
+    private final FunctionInfo info;
+
+    static {
+        DataTypes.register(TDigestStateType.ID, TDigestStateType.INSTANCE);
+    }
+
+    public static void register(AggregationImplModule mod) {
+        mod.register(NAME, new PercentileAggregationFunctionResolver());
+    }
+
+    private static class PercentileAggregationFunctionResolver implements DynamicFunctionResolver {
```

Comment:
afaik, we only use DynamicFunctionResolver if we have varargs, otherwise we just register all permutations, which should be done here too.